compiler/plugins/target/ROCM/builtins/ukernel/argmax_ukernel.c

```diff
@@ -0,0 +1,158 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <float.h>
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+
+extern "C" __device__ __attribute__((const)) half __ockl_wfred_max_f16(half);
+extern "C" __device__ __attribute__((const)) float __ockl_wfred_max_f32(float);
+extern "C" __device__ __attribute__((const))
+int64_t __ockl_wfred_min_i64(int64_t);
+extern "C" __device__ __attribute__((const))
+int32_t __ockl_wfred_min_i32(int32_t);
+
+extern "C" __device__ void
```

Comment:
These kernels assume one workgroup to handle the full reduction. We need to document it here like the KernelConfig side. Otherwise it's rather confusing to see the following just use workitem id as the index if people jump to this file directly.

Reply:
ah, good idea! added!